draft-ietf-gnap-core-protocol.md

```diff
@@ -5365,3 +5365,693 @@ justification for the data type representing the same basic kind of thing it ext
+      "properties": {
+        "jwk": {
+          "type": "object"
+        }/* this should have a JWK-specific subschema*/,
+        "cert": {
+          "type": "string"
+        },
+        "cert#S256": {
+          "type": "string"
+        },
+        "proof": {
+          "type": "string",
+          "enum": [
+            "jwsd",
+            "jws",
+            "mtls",
+            "dpop",
+            "pop",
+            "httpsig"
+          ]/* can be expanded by extensions*/
+        }
+      }
+    },
+    "subject_id_types": {
+      "type": "string",
+      "enum": [
+        "iss-sub",
+        "email",
+        "account",
+        "phone-number"
+      ]/* more added by extensions*/
+    },
+    "assertion_types": {
+      "type": "string",
+      "enum": [
+        "id_token",
+        "saml2"
+      ]/* more added by extensions*/
+    },
+    "subject_id_structures": {
+      "type": "object",
+      "required": [
+        "subject_type"
+      ],
+      "properties": {
+        "subject_type": {
+          "$ref": "#/definitions/subject_id_types"
+        }
+      },
+      "allOf": [
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "account"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "uri": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "uri"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "email"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "email": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "email"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "phone-number"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "phone_number": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "phone_number"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "subject_type": {
+                "const": "iss-sub"
+              }
+            }
+          },
+          "then": {
+            "properties": {
+              "iss": {
+                "type": "string"
+              },
+              "sub": {
+                "type": "string"
+              }
+            },
+            "required": [
+              "uri",
+              "sub"
+            ]
+          }
+        }
+      ]
+    },
+    "access_token": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "string"
```

Comment:
This guy is important enough to warrant a `format`.

Reply:
Fair point -- right now it's inherited the OAuth restrictions on access token values and so we'll need to revisit this.